src/main/java/io/cryostat/net/reports/ReportsModule.java

```diff
@@ -57,44 +56,75 @@
 
 import dagger.Module;
 import dagger.Provides;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
 
 @Module(
         includes = {
             ReportTransformerModule.class,
         })
 public abstract class ReportsModule {
 
-    static final String REPORT_GENERATION_LOCK = "REPORT_GENERATION_LOCK";
-
-    @Provides
-    @Singleton
-    @Named(REPORT_GENERATION_LOCK)
-    /** Used to ensure that only one report is generated at a time */
-    static ReentrantLock provideReportGenerationLock() {
```

Comment:
There is still a lock involved, but it isn't so explicit:

https://github.com/andrewazores/cryostat/blob/sidecar-reports/src/main/java/io/cryostat/net/reports/SubprocessReportGenerator.java#L93

The `synchronized` keyword on this method will perform the same function of only allowing one thread to cause a forked subprocess to kick off and generate a report. In the case that a sidecar generator is configured then this code path isn't hit and so the report caches should be free to call out and make multiple report requests at the same time. No lock is applied in the case of the sidecar generators because it's a desirable feature that more than one generator replica could be available behind a loadbalancer - or it could even be a loadbalancer service running a native build of the report generator as "serverless".